group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1127,7 +1176,9 @@ public void replay(
         Set<String> oldSubscribedTopicNames = new HashSet<>(consumerGroup.subscribedTopicNames());
 
         if (value != null) {
-            ConsumerGroupMember oldMember = consumerGroup.getOrMaybeCreateMember(memberId, true);
+            ConsumerGroupMember oldMember = value.instanceId() != null ?
+                    consumerGroup.getOrMaybeCreateStaticMember(memberId, value.instanceId(), true) :
+                    consumerGroup.getOrMaybeCreateMember(memberId, true);
```

Comment:
We don't need to use getOrMaybeCreateStaticMember here as we only want to look up the member by its id.